[fix] handle promise rejections for {#await} in SSR

Conduitry opened the pull request:
Fixes #6789. I haven't looked yet at how reasonable they would be to write. We aren't running any tests in Node 16 in CI, but I'm a little surprised that none of the tests involving SSR and `{#await}` are currently failing for me when running them locally. I'm not sure how else I'd cause a breakage that I could then demonstrate was fixed by this PR.

### Before submitting the PR, please make sure you do the following
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] Prefix your PR title with `[feat]`, `[fix]`, `[chore]`, or `[docs]`.
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
-  [x] Run the tests with `npm test` and lint the project with `npm run lint`

Conduitry commented:
The tests are passing without this change because of https://github.com/sveltejs/svelte/blob/3f990a96baa217aef04303afcefc47a0714748eb/test/runtime/index.ts#L28-L30

The runtime (browser) tests are listening for this event, which is preventing it from causing a crash anywhere else. I'm experimenting with having tests be responsible for detaching the `unhandledRejection` event handler when they're done.

benmccann commented on src/compiler/compile/render_ssr/handlers/AwaitBlock.ts:
```diff
@@ -13,7 +13,10 @@ export default function(node: AwaitBlock, renderer: Renderer, options: RenderOpt
 		function(__value) {
-			if (@is_promise(__value)) return ${pending};
+			if (@is_promise(__value)) {
+				__value.then(() => {}, () => {});
```
should this be `.catch` instead of `.then`?  (i'm assuming you're right and I don't understand this, but just making sure)

Conduitry replied:
It's `.then(() => {}, () => {})`, which attaches a resolve and a reject handler. This seems microscopically safer than `.catch(() => {})`, because all `is_promise` has actually confirmed is that the value is an object with a `then` method. If it's a proper promise, we should be able to use `.catch` as well, but `.then` should also be equivalent.

Conduitry commented:
The `unhandledRejection` handler that was being attached in the runtime tests was already causing other issues, it turned out. There were a couple of `test/hydration` tests that were broken but which weren't causing tests to fail. I've fixed those, made the runtime tests disconnect that event handler once they're done (which would cause SSR tests to fail without the `{#await}` fix, and which would also reveal the broken hydration tests), and also added Node 16 to the CI matrix.
